### PR TITLE
KAFKA-9991: Fix flaky unit tests

### PR DESCRIPTION
**Pull request by guozhangwang**
The latest commit #8254 on this test deleted all topics after each test, but the topic was actually shared among tests before. And after that we are relying on the less-reliable auto-topic generation to get the topic which makes the test flaky.

I'm now using different topics for different tests, also setting the app.id for tests differently.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by guozhangwang** [comment]
@mjsax @ableegoldman

File: streams/src/test/java/org/apache/kafka/streams/integration/KTableSourceTopicRestartIntegrationTest.java [hunk 1]
```diff
@@ -107,7 +109,6 @@ public void before() {
     @After
     public void after() throws Exception {
         IntegrationTestUtils.purgeLocalStreamsState(STREAMS_CONFIG);
-        CLUSTER.deleteAllTopicsAndWait(60000L);
```
Comment (abbccdda) [inline streams/src/test/java/org/apache/kafka/streams/integration/KTableSourceTopicRestartIntegrationTest.java:110]: Does the deletion logic affect the success of each test? If not, I would prefer to keep it to simplify the broker side log for each subsequent tests, such that we don't have a lot of unused topics.

File: streams/src/test/java/org/apache/kafka/streams/integration/KTableSourceTopicRestartIntegrationTest.java [hunk 2]
```diff
@@ -236,11 +237,7 @@ public void onRestoreStart(final TopicPartition topicPartition,
                                    final String storeName,
                                    final long startingOffset,
                                    final long endingOffset) {
-            try {
```
Comment (abbccdda) [inline streams/src/test/java/org/apache/kafka/streams/integration/KTableSourceTopicRestartIntegrationTest.java:239]: The original logic seems unnecessary?

**Review by abbccdda — COMMENTED** [review]
Thanks for improving the test, just minor comments.

File: streams/src/test/java/org/apache/kafka/streams/integration/KTableSourceTopicRestartIntegrationTest.java (hunk 2 above)
Comment (guozhangwang) [reply streams/src/test/java/org/apache/kafka/streams/integration/KTableSourceTopicRestartIntegrationTest.java:239]: Yes, the callee would not throw actually.

File: streams/src/test/java/org/apache/kafka/streams/integration/KTableSourceTopicRestartIntegrationTest.java (hunk 1 above)
Comment (guozhangwang) [reply streams/src/test/java/org/apache/kafka/streams/integration/KTableSourceTopicRestartIntegrationTest.java:110]: Actually I think having different topics for different test cases have a merit for debugging -- in the past I've seen we unintentionally share topics between tests which is hard to trouble-shoot from the logs. Now with different topic names we would very easily notice if something went wrong.

Comment (ableegoldman) [reply streams/src/test/java/org/apache/kafka/streams/integration/KTableSourceTopicRestartIntegrationTest.java:110]: Yeah I agree 100% with Guozhang here, I wasted a lot of time trying to debug failures in the RestoreIntegrationTest that turned out to be due to reusing a single shared topic. +1 for using different topics (especially when we actually write data to them!)

File: streams/src/test/java/org/apache/kafka/streams/integration/KTableSourceTopicRestartIntegrationTest.java [hunk 3]
```diff
@@ -52,30 +51,28 @@
...
+
     private final Time time = CLUSTER.time;
-    private KafkaStreams streamsOne;
     private final StreamsBuilder streamsBuilder = new StreamsBuilder();
     private final Map<String, String> readKeyValues = new ConcurrentHashMap<>();
 
-    private static final Properties PRODUCER_CONFIG = new Properties();
-    private static final Properties STREAMS_CONFIG = new Properties();
+    private String sourceTopic;
+    private KafkaStreams streamsOne;
```
Comment (ableegoldman) [inline streams/src/test/java/org/apache/kafka/streams/integration/KTableSourceTopicRestartIntegrationTest.java]: super nit: can we rename to `streams1` to be consistent with other tests?

File: streams/src/test/java/org/apache/kafka/streams/integration/KTableSourceTopicRestartIntegrationTest.java [hunk 4]
```diff
@@ -96,8 +93,13 @@ public static void setUpBeforeAllTests() throws Exception {
     public TestName testName = new TestName();
 
     @Before
-    public void before() {
-        final KTable<String, String> kTable = streamsBuilder.table(SOURCE_TOPIC, Materialized.as("store"));
+    public void before() throws Exception {
+        sourceTopic = SOURCE_TOPIC + "-" + testName.getMethodName();
+        CLUSTER.createTopic(sourceTopic);
+
+        STREAMS_CONFIG.put(StreamsConfig.APPLICATION_ID_CONFIG, "ktable-restore-from-source-" + testName.getMethodName());
```
Comment (ableegoldman) [inline streams/src/test/java/org/apache/kafka/streams/integration/KTableSourceTopicRestartIntegrationTest.java]: nit: use `IntegrationTestUtils#safeUniqueTestName` for the name

**Review by mjsax — APPROVED** [review]
Overall LGTM. Feel free to merge.

**Comment by guozhangwang** [comment]
Cherry-picked to 2.6.

**Pull request merged**